spec/awful/completion_spec.lua

```diff
@@ -15,6 +15,37 @@ if not has_bash and not has_zsh then
     return
 end
 
+local orig_path_env = GLib.getenv("PATH")
+local required_commands = {"bash", "zsh", "sort"}
+
+-- Change PATH
+local function get_test_path_dir()
+    local test_path = Gio.File.new_tmp("awesome-tests-path-XXXXXX")
+    test_path:delete()
+    test_path = test_path:get_path()
+    gfs.make_directories(test_path)
+
+    for _, cmd in ipairs(required_commands) do
+        local target = GLib.find_program_in_path(cmd)
+        if target then
+            local link = string.format('%s/%s', test_path, cmd)
+            Gio.File.new_for_path(link):make_symbolic_link(target)
```

Comment:
Whoops, my fault. Let's try it step-by-step this time:
```lua
local link = string.format('%s/%s', test_path, cmd)
Gio.File.new_for_path(link):make_symbolic_link(target)
```
This constructs `cmd` as a child of `test_path` and then creates a GFile instance for it. So what I should have written is `test_path:get_child(cmd):make_symbolic_link(target)`. I'll update my above comment for this.